RecoJets/JetProducers/plugins/FixedGridRhoProducerFastjetFromRecHit.cc

```diff
@@ -0,0 +1,180 @@
+  desc.add<double>("maxRapidity", 2.5);
+  desc.add<double>("gridSpacing", 0.55);
+  descriptions.add("hltFixedGridRhoProducerFastjetFromRecHit", desc);
+}
+
+FixedGridRhoProducerFastjetFromRecHit::~FixedGridRhoProducerFastjetFromRecHit() {}
+
+void FixedGridRhoProducerFastjetFromRecHit::produce(edm::Event &iEvent, const edm::EventSetup &iSetup) {
+  std::vector<fastjet::PseudoJet> inputs;
+  auto const &thresholds = iSetup.getData(ecalPFRechitThresholdsToken_);
+
+  if (skipHCAL_ && skipECAL_) {
+    throw cms::Exception("FixedGridRhoProducerFastjetFromRecHit")
+        << "skipHCAL and skipECAL both can't be True. Make at least one of them False.";
+  }
+
+  if (!skipHCAL_) {
+    for (const auto &hit : iEvent.get(hbheRecHitsTag_)) {
+      if (passedHcalNoiseCut(hit)) {
+        TLorentzVector hitp4(0, 0, 0, 0);
+        getHitP4(hit.id(), hit.energy(), hitp4, iSetup.getData(caloGeometryToken_));
+        inputs.push_back(fastjet::PseudoJet(hitp4.Px(), hitp4.Py(), hitp4.Pz(), hitp4.E()));
+      }
+    }
+  }
+
+  if (!skipECAL_) {
+    for (const auto &hit : iEvent.get(ebRecHitsTag_)) {
+      if (passedEcalNoiseCut(hit, &thresholds)) {
+        TLorentzVector hitp4(0, 0, 0, 0);
+        getHitP4(hit.id(), hit.energy(), hitp4, iSetup.getData(caloGeometryToken_));
+        inputs.push_back(fastjet::PseudoJet(hitp4.Px(), hitp4.Py(), hitp4.Pz(), hitp4.E()));
+      }
+    }
+
+    for (const auto &hit : iEvent.get(eeRecHitsTag_)) {
+      if (passedEcalNoiseCut(hit, &thresholds)) {
+        TLorentzVector hitp4(0, 0, 0, 0);
+        getHitP4(hit.id(), hit.energy(), hitp4, iSetup.getData(caloGeometryToken_));
+        inputs.push_back(fastjet::PseudoJet(hitp4.Px(), hitp4.Py(), hitp4.Pz(), hitp4.E()));
+      }
+    }
+  }
+
+  bge_.set_particles(inputs);
+  iEvent.put(std::make_unique<double>(bge_.rho()));
+}
+
+void FixedGridRhoProducerFastjetFromRecHit::getHitP4(const DetId &detId,
+                                                     float hitE,
+                                                     TLorentzVector &hitp4,
+                                                     const CaloGeometry &caloGeometry) {
+  const CaloSubdetectorGeometry *subDetGeom = caloGeometry.getSubdetectorGeometry(detId);
+  std::shared_ptr<const CaloCellGeometry> cellGeom =
+      subDetGeom != nullptr ? subDetGeom->getGeometry(detId) : std::shared_ptr<const CaloCellGeometry>();
+  if (cellGeom != nullptr) {
+    const auto &gpPos = cellGeom->repPos();
+    double thispt = hitE / cosh(gpPos.eta());
+    double thispx = thispt * cos(gpPos.phi());
+    double thispy = thispt * sin(gpPos.phi());
+    double thispz = thispt * sinh(gpPos.eta());
+    hitp4.SetPxPyPzE(thispx, thispy, thispz, hitE);
+  } else {
+    if (detId.rawId() != 0)
+      edm::LogInfo("FixedGridRhoProducerFastjetFromRecHit")
+          << "Warning : Geometry not found for a calo hit, setting p4 as (0,0,0,0)" << std::endl;
```

Comment:
```suggestion
      edm::LogWarning("FixedGridRhoProducerFastjetFromRecHit")
          << "Geometry not found for a calo hit, setting p4 as (0,0,0,0)" << std::endl;
```
Unclear to me if this should be a warning, or an "info".